tests/unit/Guzzle/MiddlewareFactoryTest.php

```diff
@@ -144,15 +145,137 @@ public function testRetryDelay() {
 			)
 		);
 
+		$delays = [];
+		$delayMocker = $this->getDelayMocker( $delays );
+
+		$handler = HandlerStack::create( $mock );
+		$handler->push( $middlewareFactory->retry( true ) );
+		$handler->push( $delayMocker );
+		$client = new Client( [ 'handler' => $handler ] );
+
+		$client->request( 'GET', '/' );
+
+		$this->assertEquals( [ 1000, 2000 ], $delays );
+	}
+
+	public function testServerErrorRetryDelay() {
+		$middlewareFactory = new MiddlewareFactory();
+
+		$mock = new MockHandler(
+			array(
+				new Response( 500 ),
+				new Response( 503 ),
+				new Response( 200 ),
+			)
+		);
```

Comment:
These test methods are quite long and I see this part repeated several times, perhaps some things can be factored out?